### PR TITLE
Hibernate Search 6.1: Extension for the outbox-polling coordination strategy

### LINKED ISSUE #22695: Hibernate Search 6.1: Extension for the outbox-polling coordination strategy
### Description

See http://hibernate.org/search/releases/6.1/#async-indexing

This feature is very interesting in a cloud environment, so I think it's important to have it in Quarkus.

Note however that the feature is currently incubating.

### Implementation ideas

I think we'll need a separate extension because the outbox-polling coordination strategy is in a separate JAR with additional dependencies (Avro in particular).

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #22695

@gsmet As discussed together, I did test locally with the Hibernate Search codestart and it works.

I will submit other PRs later to add more tests.

### CODE REVIEW

**Comment by yrodiere** [comment]
Looks like `./update-extension-dependencies.sh` no longer works now that we have relocation artifacts: it adds dependencies for the relocation artifacts...

I'll edit the result by hand.

File: docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc [hunk 1]
```diff
@@ -813,6 +813,80 @@ See link:{hibernate-search-doc-prefix}#mapper-orm-schema-management-manager[this
...
+In order to get rid of these limitations, you can
+link:{hibernate-search-doc-prefix}#architecture-examples-outbox-polling-elasticsearch[use the `outbox-polling` coordination strategy].
+This strategy creates an outbox table in the database to push entity change events to,
+and relies on a background processor to consume these events and perform automatic indexing.
+
+To enable the `outbox-polling` coordination strategy, an additional extension is required:
+
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="hibernate-search-orm-elasticsearch-aws"
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-search-orm-elasticsearch.adoc:845]: Hmmm, looks like the wrong extension.

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a small issue with the doc. Otherwise, it looks good to go.

**Comment by yrodiere** [comment]
Right, thanks. I addressed your comment and rebased on main.

**Review by gsmet — APPROVED** [review]

**Comment by yrodiere** [comment]
I think the devtools/grpc errors are unrelated, this time. @gsmet can you confirm these tests are known to be flaky?

**Comment by gsmet** [comment]
Yeah, they go berserk from time to time (and quite often these days).

**Comment by gsmet** [comment]
@yrodiere can you write a small something that I could include in the 2.7 announcement to present Hibernate Search 6.1 new features? A comment here is perfect.

**Comment by yrodiere** [comment]
Here you go. The links to the guides are dead for now, they'll only work when 2.7 becomes the default version in the documentation.

```
The Hibernate Search extension now uses Hibernate Search 6.1:

* Hibernate Search is now tested to work up to and including Elasticsearch 7.16
* Hibernate Search is now able to work with OpenSearch 1.0 to 1.2, with [just a single change in your configuration properties](https://quarkus.io/guides/hibernate-search-orm-elasticsearch#opensearch).
* Hibernate Search can now push entity change events to an outbox table in the database and coordinate between multiple instances of the application to process these events and reindex entities. This opt-in, **preview** feature allows for safer concurrent index updates with lower overhead on commit. Learn more in [this section of the guide](https://quarkus.io/guides/hibernate-search-orm-elasticsearch#coordination).
* Hibernate Search 6.1 also includes many smaller improvements, such as new search predicates or the ability to provide Elasticsearch index settings or mappings as JSON; see [the official Hibernate Search 6.1 webpage](https://hibernate.org/search/releases/6.1/) for more details.
```

**Pull request merged**